src/Route/RouteCollection.php

```diff
@@ -179,6 +181,7 @@ public function get($name)
         if ($this->has($name)) {
             $code = $this->getCode($name);
             $this->resolveElement($code);
+            \assert($this->elements[$code] instanceof Route);
```

Comment:
The method `resolveElement` should also trigger a deprecation (and throw an exception in next major) if the resolve value is not a `Route`.

```
private function resolveElement(string $code): void
{
    $element = $this->elements[$code];

    if (\is_callable($element)) {
        $resolvedElement = $element();
        if (!$resolvedElement instanceof Route) {
            ...
        }

        $this->elements[$code] = $resolvedElement;
        $this->updateCachedElement($code);
    }
}
```